### PR TITLE
Fix use of ES6 features not available in Node 4

### PR DESCRIPTION
**Pull request by simonihmig**
**tl;dr the latest release is broken in Node 4**

I got this stack trace in Travis when running FastBoot tests:
```
  ember-cli-addon-tests Unexpected token {
  ember-cli-addon-tests SyntaxError: Unexpected token {
  ember-cli-addon-tests     at exports.runInThisContext (vm.js:53:16)
  ember-cli-addon-tests     at Module._compile (module.js:373:25)
  ember-cli-addon-tests     at Object.Module._extensions..js (module.js:416:10)
  ember-cli-addon-tests     at Module.load (module.js:343:32)
  ember-cli-addon-tests     at Function.Module._load (module.js:300:12)
  ember-cli-addon-tests     at Module.require (module.js:353:17)
  ember-cli-addon-tests     at require (internal/module.js:12:17)
  ember-cli-addon-tests     at Object.<anonymous> (/tmp/d-117111-3490-141h8zz/pristine/node_modules/fastboot/src/ember-app.js:13:22)
```

This came from the use of object destructuring [here](https://github.com/ember-fastboot/fastboot/blob/master/src/fastboot-info.js#L19), which node 4 does not support. Also there were some cases of `let`/`const` used without 'use strict'.

The problem was probably uncovered with the removal of transpilation in https://github.com/ember-fastboot/fastboot/pull/121. Interestingly Travis did not catch it, because it still used Babel, probably by mistake: https://github.com/ember-fastboot/fastboot/blob/master/test/mocha.opts#L2

This PR should fix these things. 

I assume that the removal of transpilation (the alchemist thing) is still wanted, so ES6 features not supported in node 4 may not be used. In case you prefer to use the latest ES6 and bring back transpilation, I can try to change this PR accordingly. But there were so few changes required, I think it should be ok how it is now!? /cc @tomdale @rwjblue @danmcclain @arjansingh

### CODE REVIEW

**Comment by kellyselden** [comment]
@simonihmig Nice find!

**Comment by kratiahuja** [comment]
Thank you for fixing this @simonihmig !

**Comment by rwjblue** [comment]
Confirm. We want to support node 4 without transpilation.

**Comment by rwjblue** [comment]
Thanks for working on this, will try to release when I'm back to my computer...

**Comment by simonihmig** [comment]
Thanks for merging. As this will maybe break a lot of future builds involving FastBoot, maybe even production deployments, when not using a `yarn.lock`, a quick bug fix release makes sense!

@kratiahuja the breakage was caused by two separate issues, the ES6 usage (which is easy to forget about, but should be detected in the build) and the (kind of) broken test config. While identifying the first issue was easy based on the stack trace, I spent most time wondering how tests could pass (on CI as well as locally), and figuring out what's happening there. So, in other words, you should not blame yourself too much! 😉
